synfig-core/src/synfig/node.h

```diff
@@ -264,8 +264,8 @@ class Node : public etl::rshared_object
 
 	//! Callback function for a foreach method.
 	//! If it returns true, the foreach iteration is halted.
-	using ForeachFunc = sigc::slot<bool(Node*)>;
-	using ConstForeachFunc = sigc::slot<bool(const Node*)>;
+	using ForeachFunc = sigc::slot1<bool, Node*>;
+	using ConstForeachFunc = sigc::slot1<bool, const Node*>;
```

Comment:
Doesn't accept the `bool(Node*)` notation? :(

Anyway, does it need the explicit `1` in `sigc::slot`?

Reply:
Yes. For some reason gcc5 in Ubuntu 16.04 fails to compile it with the error:
```
/usr/include/sigc++-2.0/sigc++/functors/slot.h:104:19: error: function returning a function
```

Looks like it can't correct parse template with `()` in it like here: `sigc::slot<bool(Node*)>;`, so switching to this template allows us to skip this.

Reply:
OK